### ISSUE TITLE
[Feature request] dwarfs support.

### ISSUE THREAD

**Issue by Samueru-sama** [mgord9518/aisap#21]
Hi, I've recently been experimenting with the [appimage runtime](https://github.com/VHSgunzo/uruntime) that [supports dwarfs](https://github.com/AppImage/AppImageSpec/issues/36#issuecomment-2479951756).

The big oopsie I ran into is that of course aisap doesn't work with it since it expects squashfs only. 

However it is worth noting that it is possible to make aisap in its current state work with dwarfs, the problem is that it takes a horrible hack that uses [two wrapper scripts](https://i.imgur.com/Jwkq5T6.png).

So in theory all that would need to be done is check if the target appimage is dwarfs, after that the dwarfs binary can handle the rest and even find the offset on its own.

**Comment by probonopd** [comment]
Wouldn't the sensible approach to get a decision on the format in the AppImageSpec first, and then have all tools support that? Otherwise we end up with a zoo of incompatible tools outside of the spec.

**Comment by mgord9518** [comment]
@Samueru-sama The offset isn't an issue as that's something aisap finds already (as long as it's just an ELF executable that isn't compressed with UPX or something)

I was just wondering if you had added something into the main header of the file that would hint at it being different (eg changing the AppImage header). If not it shouldn't be hard to find

**Comment by mgord9518** [comment]
It should only be a couple line change then, I'll try to get it implemented soonish

**Comment by Samueru-sama** [comment]
Hi @mgord9518 are there plans for this to be added?

We recently started making squashfs releases of the [Steam AppImage](https://github.com/ivan-hc/Steam-appimage/releases/tag/20250204-041646) in part due to this issue.

**Comment by mgord9518** [comment]
@Samueru-sama I do want it as a feature, but it's been difficult for me to find the motivation currently. If you make a PR I'll merge it but otherwise I can't make any promises on time

**Comment by Samueru-sama** [comment]
> [@Samueru-sama](https://github.com/Samueru-sama) I do want it as a feature, but it's been difficult for me to find the motivation currently. If you make a PR I'll merge it but otherwise I can't make any promises on time

Understood, will see if I can study Go a bit and make something. 

Hope you are alright.

**Comment by mgord9518** [comment]
@Samueru-sama Yeah! I'm doing well, I just jump between projects. Aisap will probably become my favorite again eventually, I'm just a bit burned-out with AppImage stuff.
